src/crate/crash/command.py

```diff
@@ -322,6 +329,74 @@ def _connect(self, server):
             self.logger.info('CONNECT OK')
             self._check()
 
+    def sysinfo(self):
+        prev_format = self.output_writer.output_format
+        self._switch_format('mixed')
+        self._sysinfo()
+        self._switch_format(prev_format)
+
+    @noargs_command
+    def _sysinfo(self, *args):
+        """ print system and cluster info """
+        if self.connection.lowest_server_version >= SYSINFO_MIN_VERSION:
+            self._quantity_sysinfo()
+            self._nodes_info()
+        else:
+            tmpl = 'Crate {version} does not support the cluster "sysinfo" command'
+            self.logger.warn(tmpl.format(version=self.connection.lowest_server_version))
+
+    def _nodes_info(self):
+        self._execute(""" select version['number'] as crate_version,
+                          round(heap['max'] / 1024.0 / 1024.0)
+                                as total_heap_mb,
+                          round((mem['free'] + mem['used']) / 1024.0 / 1024.0)
+                                as total_memory_mb,
+                          os_info['available_processors'] as cpus,
+                          os['uptime'] /1000 as uptime_s,
+                          format('%s - %s (%s)',
+                                os_info['name'],
+                                os_info['version'],
+                                os_info['arch']) as os_info,
+                          format('java version \"%s\" %s %s (build %s)',
+                                os_info['jvm']['version'],
+                                os_info['jvm']['vm_vendor'],
+                                os_info['jvm']['vm_name'],
+                                os_info['jvm']['vm_version']) as jvm_info
+                          from sys.nodes
+                          order by os['uptime'] desc """)
+        self.pprint(self.cursor.fetchall(), [c[0] for c in self.cursor.description])
+
+    def _quantity_sysinfo(self):
+            self.rows = []
+            self.cols = []
+            self._num_nodes()
+            self._num_tables()
+            self._num_records_and_shards()
+            self.pprint([self.rows], self.cols)
```

Comment:
combine `_num_nodes()`, `self._num_tables()` and `self._num_records_and_shards()` into a single function call